Dockerfile.daggerd

```diff
@@ -0,0 +1,16 @@
+# syntax = docker/dockerfile:1
```

Comment:
I think we can combine this dockerfile with the other and use a different stage for daggerd (i.e. `FROM alpine as daggerd`). Not a big deal in and of itself, but if it helps clean up the code later where you have to fight between Dockerfile and Dockerfile.daggerd then we can do it now.

Reply:
I tried, and I get some weird privilege issues when using the `docker build -f` command inside go tests. All the context is here: https://github.com/dagger/dagger/pull/3187/commits/23b323b271d78ecd7e3b0580ff86362cc8866ac9